src/sonic-yang-models/yang-models/sonic-device_metadata.yang

```diff
@@ -153,6 +153,31 @@ module sonic-device_metadata {
                 leaf storage_device {
                     type boolean;
                 }
+
+                leaf asic_name {
+                    type string;
+                    description "On a VoQ switch, the ASIC Name is used as a qualifier in global
+                                 database keys to create a system wide unique key.";
+                }
+
+                leaf switch_id {
+                    type uint16;
+                    description "Vendor specific switch ID. Identifies switch chip.";
+                }
+
+                leaf switch_type {
+                    type string {
+                        pattern "chassis-packet|fabric|npu|voq";
```

Comment:
would you please add negative unit test for switch_type?

Reply:
Done.